esp-hal/src/gpio/mod.rs

```diff
@@ -1569,6 +1548,14 @@ where
         Self { pin }
     }
 
+    /// Split the pin into an input and output signal.
+    ///
+    /// Peripheral signals allow connecting peripherals together without using
+    /// external hardware.
+    pub fn split(self) -> (interconnect::InputSignal, interconnect::OutputSignal) {
```

Comment:
> we could change the pin drivers to take PeripheralInput/PeripheralOutput as other peripheral drivers

I'd rather the pin drivers accept `InputSignal`/`OutputSignal` directly rather than `PeripheralInput`, since we'd end up having do deal with nonsense like `Input::new(Level::High, Pull::None)`.

Even after saying that the pin drivers should ideally work with pins and not signals.

Reply:
> nonsense like Input::new(Level::High, Pull::None).

Haha good point, that would be unfortunate to allow.